### ISSUE TITLE
[Feature] Add Bun in benchmark

### ISSUE THREAD

**Issue by ImBIOS** [yarnpkg/berry#6559]
- [ ] I'd be willing to implement this feature ([contributing guide](https://yarnpkg.com/advanced/contributing))
- [x] This feature is important to have in this repository; a contrib plugin wouldn't do

## Describe the user story

As a developer, I often want to know which PM is currently the W, but I often need to test it out locally each time we want to find out which one is the best compared to Bun.

## Describe the solution you'd like

Add bun to benchmark

## Describe the drawbacks of your solution

It might add the tension and energy of competition inside the yarn team between another package manager, but, it's good tension and energy to compete to be the best possible.

## Describe alternatives you've considered

_none_

**Comment by zkochan** [comment]
Something to note, I was never able to reproduce locally the claimed perfomance of Bun (17x faster than pnpm and 33x faster than Yarn). Also, it wouldn't make sense to compare Bun with default configuration as they use offline cache by default. Which is an opt-in setting in all the current js-based node.js package managers.

**Comment by clemyan** [comment]
> I was never able to reproduce locally the claimed perfomance of Bun (17x faster than pnpm and 33x faster than Yarn).

Yeah, [neither can we](https://x.com/arcanis/status/1702297503465611278)

> offline cache by default. Which is an opt-in setting in all the current js-based node.js package managers.

Unless I'm mistaken about what you mean, Yarn's global cache is [on by default](https://yarnpkg.com/configuration/yarnrc#enableMirror). And, doesn't pnpm's store act as a cache?
